main_test.go

```diff
@@ -116,50 +114,39 @@ func TestEnd2EndSuccess(t *testing.T) {
 	assert.Equal(t, string(expected), string(result))
 }
 
-// TestFailMissingEnvironmentVariable ensures that the application is correctly failing
+// TestFailHierarchyMissingEnvironmentVariable ensures that the application is correctly failing
 // If an environment variable specified in `hierarchy.lst` is not found.
 // It spawns a new process to determine the exit code of the application.
 // Anything other than a 1 is a problem
 // It uses the environment variable TEST_FAIL_EMPTY to signal the actual execution of the functionality
-func TestFailMissingEnvironmentVariable(t *testing.T) {
+func TestFailHierarchyMissingEnvironmentVariable(t *testing.T) {
 	if os.Getenv("TEST_FAIL_EMPTY") == "1" {
-		var cfg config
-		cfg.hierarchyFile = "testdata/test2-with-env-fail/hierarchy.lst"
-		cfg.basePath = "testdata/test2-with-env-fail"
-		cfg.outputFile = "output.yaml"
-		cfg.filterExtension = defaultFileFilter
-		cfg.logDebug = false
-		cfg.logTrace = false
-		cfg.failMissing = false
+		cfg := cfgDefaults
```

Comment:
```suggestion
		cfg := cfgDefaults
		cfg.failMissingEnvVar = true
```
Don't we need this here?

Reply:
No, because it always fails when an env var is missing in the hierarchy. I'll add a comment.